pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -489,4 +492,40 @@ self: super: builtins.intersectAttrs super {
   # Without this override, the builds lacks pkg-config.
   opencv-extra = addPkgconfigDepend super.opencv-extra (pkgs.opencv3.override { enableContrib = true; });
 
+  # Alex has some weird files in /usr/share that create a cyclic ref with
+  # its bin dir.
+  alex = hasNoBinOutput super.alex;
+
+  # Disable separate bin outputs for these specific packages that break with it.
+  H = hasNoBinOutput super.H;
+  cryptol = hasNoBinOutput super.cryptol;
+  git-annex = hasNoBinOutput super.git-annex;
+  sproxy = hasNoBinOutput super.sproxy;
+  juandelacosa = hasNoBinOutput super.juandelacosa;
+  mywatch = hasNoBinOutput super.mywatch;
+  sugarhaskell = hasNoBinOutput super.sugarhaskell;
+  zerobin = hasNoBinOutput super.zerobin;
+
+  # Override a number of packages with specific references to $out in their
+  # derivations
+  stack = overrideCabal super.stack (drv: {
+    postInstall = ''
+      exe=$bin/bin/stack
+      mkdir -p $out/share/bash-completion/completions
```

Comment:
Bash completions should go to `$bin` (so that they get installed to the user environment the same time as the binaries).